neuro_san/http_sidecar/http_sidecar.py

```diff
@@ -102,9 +110,16 @@ def make_app(self):
     def allow(self, agent_name) -> bool:
         return self.allowed_agents.get(agent_name, False)
 
-    def update_agents(self, agents: List[str]):
+    def update_agents(self):
+        data: Dict[str, Any] = {}
+        session: ConciergeSession = DirectConciergeSession(metadata={})
+        agents_dict: Dict[str, List[Dict[str, str]]] = session.list(data)
+        agents_list: List[Dict[str, str]] = agents_dict["agents"]
+        agents: List[str] = []
+        for agent_dict in agents_list:
+            agents.append(agent_dict["agent_name"])
         with self.lock:
```

Comment:
We can do updating "allowed" agents right here, and using  DirectConciergeSession too - this is now available,
since we share process space.